clients/vault/src/oracle/agent.rs

```diff
@@ -123,24 +123,12 @@ pub async fn start_oracle_agent(
 			}
 		}
 
+		tracing::info!("start_oracle_agent(): shutting down overlay connection");
 		// shutdown the overlay connection
 		overlay_conn.stop();
 	});
 
-	tokio::spawn(on_shutdown(shutdown_sender.clone(), async move {
-		tracing::debug!("start_oracle_agent(): sending signal to shutdown overlay connection...");
-		if let Err(e) = disconnect_signal_sender.send(()).await {
-			tracing::warn!("start_oracle_agent(): failed to send disconnect signal: {e:?}");
-		}
-	}));
-
-	Ok(OracleAgent { collector, is_public_network, message_sender: Some(sender), shutdown_sender })
-}
-
-impl Drop for OracleAgent {
-	fn drop(&mut self) {
```

Comment:
Cannot use the Drop trait anymore, since `mpsc::channel::Sender` only works in async.